redis/conn.go

```diff
@@ -283,6 +308,7 @@ func DialURL(rawurl string, options ...DialOption) (Conn, error) {
 	address := net.JoinHostPort(host, port)
 
 	if u.User != nil {
+		options = append(options, DialUsername(u.User.Username()))
```

Comment:
Does this make sense if password isn't set?

Reply:
You're right. It will have no effect on the request. changed in the new commit